vscode/src/client.ts

```diff
@@ -691,9 +828,45 @@ export default class Client extends LanguageClient implements ClientInterface {
           token?: vscode.CancellationToken,
         ) => Promise<T>,
       ) => {
-        return this.benchmarkMiddleware(type, param, () =>
+        this.workspaceOutputChannel.trace(
+          `Sending request: ${JSON.stringify(type)} with params: ${JSON.stringify(param)}`,
+        );
+
+        const result = (await this.benchmarkMiddleware(type, param, () =>
           next(type, param, token),
+        )) as any;
+
+        this.workspaceOutputChannel.trace(
+          `Received response for ${JSON.stringify(type)}: ${JSON.stringify(result)}`,
         );
+
+        const request = typeof type === "string" ? type : type.method;
+
+        switch (request) {
+          case "rubyLsp/workspace/dependencies":
+            return Promise.resolve(
```

Comment:
when fn is async, you probably don't need `Promise.resolve`, you can just return